### PR TITLE
Prevent operators with many of same char to have same char after them

### PR DESCRIPTION
**Pull request by whatyouhide**
First approach at closing #7342. I am not sure the error message is the best. Let me know what you think :)

### CODE REVIEW

**Comment by whatyouhide** [comment]
@ericmj so you mean not explain that the problem is a character that follows an operator that is made of that character repeated multiple times, and just use a normal syntax error?

**Comment by ericmj** [comment]
> so you mean not explain that the problem is a character that follows an operator that is made of that character repeated multiple times, and just use a normal syntax error?

Yes, in the same sense that `**` and `//` are normal syntax errors. An explanation is not needed because it's probably a typo and not expected work so the explanation and suggestion to solve the problem using parenthesis will probably be misleading.

**Comment by whatyouhide** [comment]
@ericmj good point, I'll updated.

**Comment by whatyouhide** [comment]
Pushed a fix that emits a warning instead of erroring out. I am also taking care of `^` now. Let me know what you think of this approach.

File: lib/elixir/src/elixir_tokenizer.erl [hunk 1]
```diff
@@ -1220,3 +1225,16 @@ invalid_do_with_fn_error(Prefix) ->
...
   "Syntax error before: ".
+
+% TODO: Turn into an error on Elixir 2.0.
+maybe_warn_too_many_of_same_char([T | _] = Token, [T | _] = _Rest, Line, Scope) ->
+  Warning =
+    case T of
+      $. -> "please use parens around \"...\" instead";
+      _ -> io_lib:format("please use a space between \"~ts\" and the next \"~ts\"", [Token, [T]])
+    end,
+  Message =
```
Comment (josevalim) [inline lib/elixir/src/elixir_tokenizer.erl]: Duplicate `Message =`

Comment (whatyouhide) [reply lib/elixir/src/elixir_tokenizer.erl]: Oh good catch, how did I miss that 😄

File: lib/elixir/src/elixir_tokenizer.erl [hunk 2]
```diff
@@ -343,6 +347,7 @@ tokenize([T | Rest], Line, Column, Scope, Tokens) when T == $); T == $}; T == $]
 
 % ## Two Token Operators
 tokenize([T1, T2 | Rest], Line, Column, Scope, Tokens) when ?two_op(T1, T2) ->
+  maybe_warn_too_many_of_same_char([T1, T2], Rest, Line, Scope),
```
Comment (josevalim) [inline lib/elixir/src/elixir_tokenizer.erl]: This means we will warn on this code: `<<foo>><><<bar>>`. There is no ambiguity here. Maybe we will have to break `two_op` apart.

Comment (whatyouhide) [reply lib/elixir/src/elixir_tokenizer.erl]: Oh you're right.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**